### PR TITLE
Fix providing multiple environment variables on windows (#267)

### LINKED ISSUE #267: Calling program with different environment variables on Windows doesn't work
Calling a program on Windows with a modified environment does not work if more than a single environment variable is provided (processx version 3.4.3). Instead, the first environment variable is duplicated. In the following example, `ASD` is set twice (to the value `def`) and `BLA` is not set at all. Note that this only occurs on Windows.

Reprex:

``` r
processx::run(file.path(R.home("bin"), "R"), c("-e", "write(Sys.getenv('BLA'), stderr())"), 
    env = c(ASD = "def", BLA = "blub"))$stderr
#> [1] "\r\n"
```

### PR DESCRIPTION
**Pull request by rschuchmann**
Fixes #267

### CODE REVIEW

File: tests/testthat/test-env.R [hunk 1]
```diff
@@ -19,16 +19,16 @@ test_that("inherit by default", {
 
 test_that("specify custom env", {
 
-  v <- basename(tempfile())
+  v <- c(basename(tempfile()), basename(tempfile()))
   if (os_type() == "unix") {
-    cmd <- c("bash", "-c", paste0("echo $", v))
+    cmd <- c("bash", "-c", paste0("echo $", paste(v, collapse = " $")))
```
Comment (gaborcsardi) [inline tests/testthat/test-env.R:24]: I think this is more readable: `paste0("echo ", paste0("$", v, collapse = " "))`.

File: tests/testthat/test-env.R [hunk 2]
```diff
@@ -19,16 +19,16 @@ test_that("inherit by default", {
...
   } else {
-    cmd <- c("cmd",  "/c", paste0("echo %", v, "%"))
+    cmd <- c("cmd",  "/c", paste0("echo %", paste(v, collapse = "% %"), "%"))
```
Comment (gaborcsardi) [inline tests/testthat/test-env.R:26]: Similarly, I think `paste0("echo ", paste0("%", v, "%", collapse = " "))` is more readable here.

**Review by gaborcsardi — COMMENTED** [review]
Thanks! Looks great, can you please add a bullet point to the NEWS?

**Comment by rschuchmann** [comment]
> 
> 
> Thanks! Looks great, can you please add a bullet point to the NEWS?
Done. I have also accepted your suggestions in the tests.

**Comment by gaborcsardi** [comment]
Great! Thanks much!

**Pull request merged**